### DISCUSSION TITLE
Extract the direct link apk at this page

### DISCUSSION THREAD

**Discussion by linux40** [ImranR98/Obtainium#1523] [Q&A]
I want obtainium to handle the following page:
https://s1fun.koalcat.com/app/
There is only one direct link to apk. The link points to the same host as the page, but it presents as the full link:
```
<a class="download_android__2lJIG" href="https://s1fun.koalcat.com/app/s1fun-1.4.0-build57.apk" rel="noopener noreferrer" target="_blank"></a>
```
I tried to use HTML source to handle the page, but the best I can achieve is to get a `noVersionFound` error.

For example, I set `([.\w]+-\w+)\.apk` as the version pattern, and enabled `Apply version string extraction Regex to entire page`, but this does not help. I also set the pattern for apk file to `.*`, otherwise, I will get the `noReleaseFound` error (even with the default `.apk$`).

I guess I can use direct-apk-link source. However, I prefer this page to check updates. Any idea? Thanks in advance.

**Comment by ImranR98** [comment]
The site is blank with JavaScript disabled (Obtainium can't run JS). You should still be able to use the direct APK link source. As long as Obtainium can detect that the first 1kb of the APK is consistent on each download request, you should still get update notifications (if not, it won't let you add the app anyways).

**Reply by linux40** [reply]
Thanks. I will try direct link for a while.

**Reply by ImranR98** [reply]
Sorry, just noticed the app version number is part of the link. That means the direct APK method won't be usable since the link itself changes.
